### PR TITLE
chore(schema): enable linting

### PR DESCRIPTION
**Pull request by DamianGlowala**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR enables linting for `packages/schema` and applies suggestions, as well as addresses errors resulting from this change.

closes https://github.com/nuxt/nuxt/pull/22544

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 dist
 node_modules
-schema
 **/*.tmpl.*
```
Comment (danielroe) [inline .eslintignore:3]: I think this was meant to ignore `packages/schema/schema` but of course we do want to include `packages/schema` - maybe refine it instead?

```suggestion
schema/schema
**/*.tmpl.*
```

Comment (DamianGlowala) [reply .eslintignore:3]: Good point! I'll add `packages/schema/schema` for explicitness to the `.eslintignore` (even though it only contains .d.ts, .md and .json, which are not linted, but surely it's better not to rely on file extensions).

File: packages/schema/src/types/nuxt.ts [hunk 2]
```diff
@@ -4,6 +4,57 @@ import type { NuxtHooks, NuxtLayout, NuxtMiddleware } from './hooks'
...
+    ssr?: boolean
+    src: string
+    mode?: 'all' | 'server' | 'client'
+    /**
+     * This allows more granular control over plugin order and should only be used by advanced users.
+     * Lower numbers run first, and user plugins default to `0`.
+     *
+     * Default Nuxt priorities can be seen at [here](https://github.com/nuxt/nuxt/blob/9904849bc87c53dfbd3ea3528140a5684c63c8d8/packages/nuxt/src/core/plugins/plugin-metadata.ts#L15-L34).
+     */
+    order?: number
```
Comment (danielroe) [inline packages/schema/src/types/nuxt.ts:18]:
```suggestion
  /** @deprecated use mode */
  ssr?: boolean
  src: string
  mode?: 'all' | 'server' | 'client'
  /**
   * This allows more granular control over plugin order and should only be used by advanced users.
   * Lower numbers run first, and user plugins default to `0`.
   *
   * Default Nuxt priorities can be seen at [here](https://github.com/nuxt/nuxt/blob/9904849bc87c53dfbd3ea3528140a5684c63c8d8/packages/nuxt/src/core/plugins/plugin-metadata.ts#L15-L34).
   */
  order?: number
```

Comment (DamianGlowala) [reply packages/schema/src/types/nuxt.ts]: Wondering the same, as I tried changing quotes to double quotes, and it correctly detected the issue. Looking at the `ssr`'s indent, for some reason it seems to be fine...

image

Comment (DamianGlowala) [reply packages/schema/src/types/nuxt.ts]: It appears to have something to do with the folder name `types` this file is in.

**Pull request merged**